crates/iota-sdk/src/iota_client_config.rs

```diff
@@ -79,17 +79,18 @@ impl IotaClientConfig {
     }
 
     /// Get the first [`IotaEnv`] or one by its alias.
-    pub fn get_env(&self, alias: &Option<String>) -> Option<&IotaEnv> {
-        if let Some(alias) = alias {
-            self.envs.iter().find(|env| &env.alias == alias)
-        } else {
-            self.envs.first()
-        }
+    pub fn get_env(&self, alias: &str) -> Option<&IotaEnv> {
+        self.envs.iter().find(|env| &env.alias == alias)
     }
 
     /// Get the active [`IotaEnv`].
     pub fn get_active_env(&self) -> Result<&IotaEnv, anyhow::Error> {
-        self.get_env(&self.active_env).ok_or_else(|| {
+        if let Some(active_env) = &self.active_env {
+            self.get_env(active_env)
+        } else {
+            self.envs.first()
+        }
```

Comment:
should we also set `self.active_env` to `self.envs.first()` then? Feels weird that we claim to have an active env but the internal state of the object is different.

Reply:
It's not a mutable fn and I don't think it should be

Reply:
Sure, but the API is kind of lying to the user, because the object's internal state doesn't reflect what the method returns.

Reply:
I don't think that's true. If there's no env the default is the first, which is represented in the state